test/reqs.test.ts

```diff
@@ -0,0 +1,207 @@
+import * as hre from "hardhat";
+import { ethers } from "ethers";
+import { expect } from "chai";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import {
+  GameVault,
+  Games,
+  MockERC20,
+  MockERC721,
+  ObjectRegistry,
+  ObjectRegistry__factory,
+  Seasons,
+  StakerRewards,
+  XP,
+} from "../typechain";
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+
+// Test core requirements for Staking
+
+// Priority #1 - Staking
+// 1.1 - User visits a staking website (possibly a zApp).
+// 1.2 - User can stake their NFT (e.g., Wilder Wheel)
+// 1.3 - User receives staked NFT in return
+// 1.4 - User receives a Race Pass (new contract)
+// 1.5 - User receives a percentage of rewards on an epoch (passive rewards)
+// 1.6 - User can unstake at any time.
+
+describe("Requirements testing", () => {
+  let deployer: SignerWithAddress;
+  let staker: SignerWithAddress;
+
+  let mockERC20: MockERC20;
+  let mockERC721: MockERC721;
+
+  let games: Games;
+  let gameVault: GameVault; // ERC721Wrapper
+  let rewards: StakerRewards;
+  let seasons: Seasons;
+  let xp: XP; // is ERC20
+
+  let stakingGameRegistryAddress: string;
+  let stakingGameRegistry: ObjectRegistry;
+
+  let seasons0RegistryAddress: string;
+  let seasons0Registry: ObjectRegistry;
+
+  let seasonsRegistry: ObjectRegistry;
+  let gameVaultRegistry: ObjectRegistry;
+  let xpRegistry: ObjectRegistry
+
+  // Setup game
+  before(async () => {
+    [
+      deployer,
+      staker,
+    ] = await hre.ethers.getSigners();
+
+    // Deployments
+    // 1. MockERC20
+    // 2. MockERC721
+    // 3. Games
+    // 4. Seasons
+    // 5. XP
+    // 6. GameVault
+    // 7. StakerRewards
+
+    const mockERC20Factory = await hre.ethers.getContractFactory("MockERC20");
+    // Will immediately mint and transfer to msg.sender
+    mockERC20 = await mockERC20Factory.deploy("MEOW", "MEOW");
+
+    const mockERC721Factory = await hre.ethers.getContractFactory("MockERC721");
+    mockERC721 = (await mockERC721Factory.deploy("WilderWheels", "WW", "0://wheels-base"));
+
+    const gamesFactory = await hre.ethers.getContractFactory("Games");
+    games = await gamesFactory.deploy();
+
+    const gameName = "StakingGame";
+    const gameNameBytes = hre.ethers.utils.formatBytes32String(gameName);
+    await games.createGame(gameNameBytes, deployer.address, "a-staking-game");
+
+    // Registry for the StakingGame
```

Comment:
Yes you successfully get the gameObjects here, which is the ObjectRegistry for the game you're in.

Reply:
(worth pointing out as this is a complicated part of the system. it is one step removed from just inheriting the ObjectRegistry in Games.sol, the simplest and first way I did it)